python/helpers.py

```diff
@@ -0,0 +1,172 @@
+
+
+def read_master_key(path="./master-key.txt"):
+    return Binary(open(path, "rb").read(96))
+
+
+def create_json_schema(data_key=None):
+    if data_key is None:
+        raise ValueError("data_key must be provided")
+
+    patient_schema = """
+    {{
+        "medicalRecords.patients": {{
+            "bsonType": "object",
+            "encryptMetadata": {{
+                "keyId": [
+                    {{
+                        "$binary": {{
+                            "base64": "{data_key}",
+                            "subType": "04"
+                        }}
+                    }}
+                ]
+            }},
+            "properties": {{
+                "insurance": {{
+                    "bsonType": "object",
+                    "properties": {{
+                        "policyNumber": {{
+                            "encrypt": {{
+                                "bsonType": "int",
+                                "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+                            }}
+                        }}
+                    }}
+                }},
+                "medicalRecords": {{
+                    "encrypt": {{
+                        "bsonType": "array",
+                        "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+                    }}
+                }},
+                "bloodType": {{
+                    "encrypt": {{
+                        "bsonType": "string",
+                        "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+                    }}
+                }},
+                "ssn": {{
+                    "encrypt": {{
+                        "bsonType": "int",
+                        "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+                    }}
+                }}
+            }}
+        }}
+    }}
+    """.format(data_key=data_key)
+
+    return json_util.loads(patient_schema)
+
+
+class ClientBuilder:
+    def __init__(self,
+                 kms_providers=None,
+                 key_alt_name="demo-data-key",
+                 key_db="encryption",
+                 key_coll="__keyVault",
+                 schema=None,
+                 connection_string="mongodb://localhost:27017",
+                 mongocryptd_bypass_spawn=False,
+                 mongocryptd_spawn_path="mongocryptd"):
+
+        super().__init__()
+        if kms_providers is None:
+            raise ValueError("kms_provider is required")
+        self.kms_providers = kms_providers
+        self.key_alt_name = key_alt_name
+        self.key_db = key_db
+        self.key_coll = key_coll
+        self.key_vault_namespace = f"{key_db}.{key_coll}"
+        self.schema = schema
+        self.connection_string = connection_string
+        self.mongocryptd_bypass_spawn = mongocryptd_bypass_spawn
+        self.mongocryptd_spawn_path = mongocryptd_spawn_path
+
+        fle_opts = AutoEncryptionOpts(
+            self.kms_providers,
```

Comment:
This line is not consistent with line 89 in which you're using the values of the params vs the instance fields. Would it be helpful for the user to stay consistent?